sklearn/utils/estimator_checks.py

```diff
@@ -97,7 +97,27 @@
 REGRESSION_DATASET = None
 
 
+def _raise_for_missing_tags(estimator, tag_name, Mixin):
+    tags = get_tags(estimator)
+    estimator_type = Mixin.__name__.replace("Mixin", "")
+    if getattr(tags, tag_name) is None:
+        raise RuntimeError(
+            f"Estimator {estimator.__class__.__name__} seems to be a {estimator_type},"
+            f" but the `{tag_name}` tag is not set. Either set the tag manually"
+            f" or inherit from the {Mixin.__name__}. Note that the order of inheritance"
+            f" matters, the {Mixin.__name__} should come before BaseEstimator."
+        )
+
+
 def _yield_api_checks(estimator):
+    if not isinstance(estimator, BaseEstimator):
```

Comment:
Did we made the decision to only really support estimators that inherit from `BaseEstimator`? If so, is it documented somewhere so third-party estimator developers can know.

Reply:
Yes, documented: 

https://github.com/scikit-learn/scikit-learn/blob/296aba8533de4e4801c94cf4dd45919de30fc48e/doc/developers/develop.rst?plain=1#L95-L98

And the example:

https://github.com/scikit-learn/scikit-learn/blob/296aba8533de4e4801c94cf4dd45919de30fc48e/doc/developers/develop.rst?plain=1#L313-L344